pkg/app/reconcile_timer.go

```diff
@@ -39,20 +39,27 @@ func (rt ReconcileTimer) IsReadyAt(timeAt time.Time) bool {
 		return false
 	}
 
-	// Did we deploy at least once?
-	lastFetch := rt.app.Status.Fetch
-	if lastFetch == nil {
-		return true
+	var lastReconcileTime time.Time
+
+	// Use latest deploy time if available, otherwise fallback to fetch
+	// If no timestamp is available, enqueue immediately
+	if rt.app.Status.Deploy != nil {
+		lastReconcileTime = rt.app.Status.Deploy.UpdatedAt.Time
```

Comment:
hmm, i wonder if we should be defensive here and check if UpdatedAt is zero time. wdyt?

Reply:
Is this even possible? Looks like any time we issue an update to the status we are also getting the time, so they should be either nil or have a time. I dont know that adding in the check would hurt anything so happy to do it.

Reply:
i was just thinking it's a bit hard to keep this invariant true. should we be defensive here to avoid accidently hammering kapp-controller if we screw up somewhere?

Reply:
I'm not sure what the check would do here though. If we return true because it is 0, it is the same as returning true because currTime - 0 >= 30s, which will always be true. If we return false, we will never run a reconcile loop because the status will never be updated and the time will always be 0, so users would have to unstick the app by updating it somehow.

Reply:
we would just not use deploy section, fallback to fetch.